chips/sifive/src/uart.rs

```diff
@@ -117,7 +119,11 @@ impl<'a> Uart<'a> {
             rx_len: Cell::new(0),
             rx_position: Cell::new(0),
             rx_status: Cell::new(UARTStateRX::Idle),
-        }
+        };
+
+        // Make sure current_baud matches what the hardware is actually configured for
+        result.set_baud_rate(115200);
```

Comment:
Is it OK to execute a function while creating a Uart object? I am doing this just to synchronize the hardware state with what the software is assuming (specifically, the `current_baud` new variable).

Reply:
I don't think this is right, the caller should specify the baud rate

Reply:
OK. In the latest code, I changed it so that the system clock is provided when the UART object is created, alleviating the issue this is trying to fix.